From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 2 Dec 2024 16:51:04 -0600
Subject: [PATCH] solus: Build libs shared

They changed these libs to be static-only upstream purely because of issues with the Qt installer, but those don't affect us so let's just keep them shared.

---
 src/jsonrpc/CMakeLists.txt                  | 2 +-
 src/jsonrpc/qtjsonrpcglobal.h               | 2 +-
 src/languageserver/CMakeLists.txt           | 2 +-
 src/languageserver/qtlanguageserverglobal.h | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/jsonrpc/CMakeLists.txt b/src/jsonrpc/CMakeLists.txt
index c2950b4..f960325 100644
--- a/src/jsonrpc/CMakeLists.txt
+++ b/src/jsonrpc/CMakeLists.txt
@@ -7,7 +7,6 @@
 
 qt_internal_add_module(JsonRpcPrivate
     INTERNAL_MODULE
-    STATIC
     SOURCES
         qhttpmessagestreamparser_p.h qhttpmessagestreamparser.cpp
         qjsonrpcprotocol.cpp qjsonrpcprotocol_p.h qjsonrpcprotocol_p_p.h
@@ -20,4 +19,5 @@ qt_internal_add_module(JsonRpcPrivate
         QT_NO_CONTEXTLESS_CONNECT
     PUBLIC_LIBRARIES
         Qt::Core
+    NO_GENERATE_CPP_EXPORTS
 )
diff --git a/src/jsonrpc/qtjsonrpcglobal.h b/src/jsonrpc/qtjsonrpcglobal.h
index 7c23910..199b021 100644
--- a/src/jsonrpc/qtjsonrpcglobal.h
+++ b/src/jsonrpc/qtjsonrpcglobal.h
@@ -8,6 +8,6 @@
 
 // include qtjsonrpcexports.h here instead
 // if module is no longer unconditionally static
-#define Q_JSONRPC_EXPORT
+#define Q_JSONRPC_EXPORT Q_DECL_EXPORT
 
 #endif // QTJSONRPCGLOBAL_H
diff --git a/src/languageserver/CMakeLists.txt b/src/languageserver/CMakeLists.txt
index e45916c..0242739 100644
--- a/src/languageserver/CMakeLists.txt
+++ b/src/languageserver/CMakeLists.txt
@@ -7,7 +7,6 @@
 
 qt_internal_add_module(LanguageServerPrivate
     INTERNAL_MODULE
-    STATIC
     SOURCES
       qlanguageserverprespectypes_p.h
       qlanguageserverspec_p.h
@@ -24,6 +23,7 @@ qt_internal_add_module(LanguageServerPrivate
     PUBLIC_LIBRARIES
         Qt::CorePrivate
         Qt::JsonRpcPrivate
+    NO_GENERATE_CPP_EXPORTS
 )
 
 if(MSVC)
diff --git a/src/languageserver/qtlanguageserverglobal.h b/src/languageserver/qtlanguageserverglobal.h
index 872960c..38268f1 100644
--- a/src/languageserver/qtlanguageserverglobal.h
+++ b/src/languageserver/qtlanguageserverglobal.h
@@ -7,6 +7,6 @@
 
 // include qtlanguageserverexports.h here instead
 // if module is no longer unconditionally static
-#define Q_LANGUAGESERVER_EXPORT
+#define Q_LANGUAGESERVER_EXPORT Q_DECL_EXPORT
 
 #endif // QTLANGUAGESERVERGLOBAL_H
